### PR TITLE
Make Config's setDir() fluent like the rest of methods

### PR DESCRIPTION
**Pull request by gonzaloserrano**

### CODE REVIEW

File: Symfony/CS/Config/Config.php [hunk 1]
```diff
@@ -51,6 +51,8 @@ public static function create()
     public function setDir($dir)
     {
         $this->dir = $dir;
+
+        return $this;
```
Comment (GrahamCampbell) [inline Symfony/CS/Config/Config.php]: You need to update the phpdoc on the interface.

**Comment by keradus** [comment]
Why have you closed it @gonzaloserrano ?

Change itself was good, just need to update docs ;)

**Comment by gonzaloserrano** [comment]
Hey, not sure, i thought it would break some ConfigInterface implementation but maybe i was sleepy. Will open again.

**Comment by keradus** [comment]
Yes, it would break public interface, therefore you send this PR not for our stable 1.x line, but for next 2.x line ;) So everything is good ;)

**Comment by GrahamCampbell** [comment]
I think with would be fine in 1.6. Yeh, it is technically a break, though it won't actually cause php errors unlike adding a param to a function defined in an interface.

**Comment by ceeram** [comment]
I dont see this as BC breaking change either. Previously the method didnt return anything, so no one will depend on the return value.

**Comment by GrahamCampbell** [comment]
It's only breaking in the sense that existing implementations of this interface no longer return the correct value after this change (ie, implementations outside this repo).

**Comment by ceeram** [comment]
There was no return value, So no outside repo will depend on the returned
value.

2015-03-11 21:53 GMT+01:00 Graham Campbell <notifications@github.com>:

**Comment by ceeram** [comment]
So technically it breaks BC, realisticly not

**Comment by keradus** [comment]
Our PHP CS Fixer has some interface at 1.x.
Let say some create a repo with additions to our stable 1.x and create some additional Config classes.

If we merge it into 1.x line, then we will start using this new interface.
Then, someone who so our 1.x interface after that change will start to using it too. And that is cool. But finally he will also see that repo with additional configs. PHP CS Fixer at 1.x level, and additional configs for 1.x level. Great. But it won't work together. Why? Due to that interface change.

If something is a BC break in technical aspect, it is a BC break indeed.

**Pull request closed without merging**